Examples/distributed_training.py

```diff
@@ -0,0 +1,132 @@
+from typing import Optional, Tuple
+from xcube.core.store import new_data_store
+from global_land_mask import globe
+from torch.utils.data import random_split
+import dask.array as da
+from torch import nn
+from mltools.distributed_training import ddp_init, prepare_dataloader, dist_train, Trainer
+import torch
+from torch.utils.data import Dataset
+import numpy as np
+from mltools.cube_utilities import iter_data_var_blocks, get_chunk_by_index, get_chunk_sizes
+from mltools.sampling import assign_block_split
+from mltools.data_processing import standardize, get_statistics
+
+
+class ChunkDataset(Dataset):
```

Comment:
Why is it required to derive from `torch.utils.data.Dataset`? If it is really required, I suggest [composition over inheritance](https://python-patterns.guide/gang-of-four/composition-over-inheritance/)

Reply:
The necessity for inheritance largely depends on the specific use case at hand. It becomes particularly valuable when dealing with scenarios where the complete dataset does not fit into memory, and we aim to avoid interrupting parallel training processes with potential restarts. Given this context, it is essential to transition from employing a DataLoader to utilizing a ChunkLoader. This approach facilitates handling data in segments rather than loading it in traditional batches, which can be more efficient under these circumstances.

For now I reduced the dataset to a size not requiring inheritance. In the future we can consider developing a dedicated class specifically tailored for this purpose. This would enable users to leverage these capabilities without dealing with inheritance.